pkg/precision/precision.go

```diff
@@ -0,0 +1,34 @@
+package precision
+
+import (
+	"math"
+
+	"github.com/shopspring/decimal"
+)
+
+func ToSignificant(input decimal.Decimal, sigDigits int32, maxScale int32) decimal.Decimal {
```

Comment:
from the YIP:
- All API input shall return an error for orders where prices or amounts exceed 18 and 8 precision respectfully

I suggest having a helper, that will check the provided decimal for the given scale, the helper should be exported to be able to be used in the API from this package